OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,299 @@
+package com.optimove.android.preferencecenter;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.optimove.android.Optimove;
+import com.optimove.android.OptimoveConfig;
+import com.optimove.android.PreferenceCenterConfig;
+import com.optimove.android.main.common.UserInfo;
+import com.optimove.android.main.tools.networking.HttpClient;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class OptimovePreferenceCenter {
+    private static final String TAG = OptimovePreferenceCenter.class.getName();
+    private static OptimovePreferenceCenter shared;
+    private static OptimoveConfig config;
+
+    /**
+     * package
+     */
+    static ExecutorService executorService;
+    /**
+     * package
+     */
+    static final Handler handler = new Handler(Looper.getMainLooper());
+
+    public interface PreferencesGetHandler {
+        void run(@Nullable Preferences preferences);
+    }
+
+    public interface PreferencesSetHandler {
+        void run(Boolean result);
+    }
+
+    public enum Channel {
+        MOBILE_PUSH(489), WEB_PUSH(490), SMS(493), IN_APP(427), WHATSAPP(498), MAIL(15), INBOX(495);
+        private final int channel;
+
+        Channel(int channel) {
+            this.channel = channel;
+        }
+
+        @NonNull
+        public int getValue() {
+            return channel;
+        }
+
+        private static Channel getChannelByValue(int value) {
+            switch (value) {
+                case 489:
+                    return MOBILE_PUSH;
+                case 490:
+                    return WEB_PUSH;
+                case 493:
+                    return SMS;
+                case 427:
+                    return IN_APP;
+                case 498:
+                    return WHATSAPP;
+                case 15:
+                    return MAIL;
+                case 495:
+                    return INBOX;
+                default:
+                    throw new IllegalArgumentException("Preference center does not support channel " + value);
+            }
+        }
+    }
+
+    public static OptimovePreferenceCenter getInstance() {
+        if (shared == null) {
+            throw new IllegalStateException("OptimovePreferenceCenter is not initialized");
+        }
+        return shared;
+    }
+
+    /**
+     * Initializes an instance of OptimovePreferenceCenter
+     *
+     * @param currentConfig current config
+     */
+    public static void initialize(OptimoveConfig currentConfig) {
```

Comment:
this must stay public to call from different namespace, but it's not for customers. can add a comment above smth like 

`This API is intended for internal SDK use. Do not call this API or depend on it in your app.`